### PR TITLE
Feat/add haptics to Drawer

### PR DESCRIPTION
**Pull request by lidord-wix**
## Description
Add haptic feedback to the Drawer component
Jira 1201

Depending on https://github.com/wix/react-native-ui-lib/pull/1277 

## Changelog
Add haptic feedback to the Drawer component

### CODE REVIEW

File: src/components/drawer/Swipeable.tsx
Comment (Inbal-Tish) [inline src/components/drawer/Swipeable.tsx:129]: I think we should keep the haptic optional, not all users will want to use it.

File: src/components/drawer/index.tsx
Comment (Inbal-Tish) [inline src/components/drawer/index.tsx:103]: This is a breaking change. You can't just delete props, you need to deprecate them. Give the user a chance to migrate.

Comment (lidord-wix) [reply src/components/drawer/index.tsx:103]: Right, adding it to the V7 spreadsheet

File: src/components/drawer/Swipeable.tsx
Comment (Inbal-Tish) [reply src/components/drawer/Swipeable.tsx:130]: Why not get the haptic method from the HapticService, i.e. `HapticMethods.impactMedium`?

Comment (lidord-wix) [reply src/components/drawer/Swipeable.tsx:130]: I'll be able to do that only when I'll fix the `HapticMethods` in the `HapticService` as we talked about in our dev sync.
I'm waiting for this PR to be merged to do so

Comment (Inbal-Tish) [reply src/components/drawer/Swipeable.tsx:130]: Ok. Add the other PR as a dependency and we'll merge this after it.

File: src/components/drawer/Swipeable.tsx
Comment (Inbal-Tish) [inline src/components/drawer/Swipeable.tsx:129]: this line repeats 3 times. How about moving it to a method?

File: src/components/drawer/Swipeable.tsx
Comment (Inbal-Tish) [inline src/components/drawer/Swipeable.tsx:19]: Is this working? The other PR merged?

Comment (lidord-wix) [reply src/components/drawer/Swipeable.tsx:19]: It is not merged yet - I added a link in the description.
And yes it works, I checked it..

Comment (Inbal-Tish) [reply src/components/drawer/Swipeable.tsx:19]: If you're using HAPTIC_METHOD only once, why store it in a const?

File: src/components/drawer/Swipeable.tsx
Comment (Inbal-Tish) [inline src/components/drawer/Swipeable.tsx:260]: Why is there a comma here? Doesn't it fail on lint tests?

File: src/components/drawer/Swipeable.tsx
Comment (Inbal-Tish) [reply src/components/drawer/Swipeable.tsx:19]: Also, why `impactMedium` is the default?

File: src/components/drawer/Swipeable.tsx
Comment (lidord-wix) [reply src/components/drawer/Swipeable.tsx]: No, but I removed it.

File: src/components/drawer/Swipeable.tsx
Comment (lidord-wix) [reply src/components/drawer/Swipeable.tsx]: These are the UX guidelines.
We can open it to a discussion in the future if any user will ask to change it.

Comment (Inbal-Tish) [reply src/components/drawer/Swipeable.tsx]: I'm asking because our private config (Drawer.config) is setting it to `impactLight`, so it should be the default or it should be open for user's change.

File: src/components/drawer/Swipeable.tsx
Comment (Inbal-Tish) [reply src/components/drawer/Swipeable.tsx:133]: Is the dependency merged?

Comment (lidord-wix) [reply src/components/drawer/Swipeable.tsx:133]: Yes :)

File: src/components/drawer/Swipeable.tsx
Comment (lidord-wix) [reply src/components/drawer/Swipeable.tsx]: It has updated to `impactMedium`, which is the default now.

**Review by Inbal-Tish — APPROVED** [review]

**Pull request merged**
